test/functional/test_framework/script_util.py

```
@@ -25,32 +33,43 @@
     script = check_script(script)
-    return scripthash_to_p2sh_script(hash160(script), main)
+    return scripthash_to_p2sh_script(hash160(script))
+
 
 def check_key(key):
     if isinstance(key, str):
-        key = bytes.fromhex(key) # Assuming this is hex string
+        key = bytes.fromhex(key)  # Assuming this is hex string
     if isinstance(key, bytes) and (len(key) == 33 or len(key) == 65):
         return key
     assert False
```

Ruff (0.8.2)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)